_posts/2020-03-17-junior-dev-advice.markdown

```diff
@@ -0,0 +1,121 @@
+that technical competency, no matter how high level, is still a
+commodity - no matter how good you are, there are plenty of other
+developers out there who are at least that good. Thus the old saw
+about it not being what you know, but who you know, is still true even
+in a supposedly knowledge oriented field. Also, keeping your skills
+up to date will mean more than a few conferences and user group
+meetups throughout your career - events that are, by default, social
+events. Be prepared to do a lot of (human to human) networking.
+
+On a more day to day level, a developer who can't communicate clearly
+and professionally with their peers and co-workers is not going to
+keep their job for long. Good developers command good salaries, but
+good salaries always come with expectations attached, and one of those
+will always be that you can work well in a team and communicate with
+others. If someone's going to pay you all of that money, they'll
+expect a lot of transparency in return. And that expectation will
+only grow as you acquire more and more experience. Your experience
+will have value, and your employers will expect you to share that
+value with those around you  by coaching and mentoring teammates,
+*without* stepping on any toes in the process.
+
+Some more recommended reads:
+
+- [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+- [Technical Debt Isn't Just
+Technical](https://corgibytes.com/blog/2020/02/12/technical-debt-isnt-just-technical/)
+(communication solves a lot of problems up front!)
+- [The Developer’s Edge: How To Become A Senior
+Developer](https://medium.com/zerotomastery/developers-edge-how-to-become-a-senior-developer-f1ec1738cf45)
+ (notice how few of these skills are technical!)
+
+# Stay Humble
+
+It won't take long for a newly minted developer to learn that there
+are plenty of developers who have a bit of an ego problem. Aside from
+the obvious problems that this can cause with teamwork and
+professional behavior, losing perspective in this way can lead to
+skills starting to stagnate. When you take on a job as a developer,
+whether you realize it or not, you're implicitly taking on another
+responsibility - the responsibility of always being a student. The
+biggest difference between a newbie developer and a seasoned veteran
+is that the senior developer *has been a student for a lot longer.*
+But remaining a life long student requires you to understand the
+difference between confidence and ego, and commit to putting aside
+even the slightest shred of ego. Stay open minded, and willing to
+consider the possibility that your previous understanding of something
+was incomplete or mistaken. Recognize that no matter how much you've
+learned, it's only the base for the next thing that you're going to
+learn. And realize that no matter how far you've come, you can still
+make dumb mistakes - and program accordingly. In a field full of big
```

Comment:
@rdormer In fact, I like to say that "if you are not making mistakes, you are not learning fast enough."

You can take it or leave it. :)